components/TextFields/MDCTextInput.h

```diff
@@ -0,0 +1,182 @@
+ above the input when text has been entered.
+ https://material.io/guidelines/components/text-fields.html#text-fields-input
+ */
+@property(nonatomic, nullable, copy) IBInspectable NSString *placeholder;
+
+/**
+ The color applied to the placeholder when inline (not floating.)
+
+ Default is black with Material Design hint text opacity.
+ */
+@property(nonatomic, nullable, strong) UIColor *inlinePlaceholderColor NS_SWIFT_NAME(inlineScale)
+    UI_APPEARANCE_SELECTOR;
+
+/**
+ The color applied to the placeholder when floating. However, when in error state, it will be
+ colored with the error color.
+
+ Default is black with Material Design hint text opacity.
+ */
+@property(nonatomic, nullable, strong)
+    UIColor *floatingPlaceholderColor NS_SWIFT_NAME(floatingColor) UI_APPEARANCE_SELECTOR;
+
+/**
+ The font applied to placeholder labels.
+
+ The UIFont.pointSize is respected for placeholders in an inline mode. However, the pointSize of
+ a placeholder in floating mode is placeholderFont.pointSize * floatingPlaceholderScale.
+
+ Default is system body.
+ */
+@property(nonatomic, nullable, strong) UIFont *placeholderFont UI_APPEARANCE_SELECTOR;
+
+/**
+ The scale of the the floating placeholder label in comparison to the inline placeholder specified
+ as a value from 0.0 to 1.0.
+
+ Default is 0.75.
+ */
+@property(nonatomic, assign) CGFloat floatingPlaceholderScale NS_SWIFT_NAME(floatingScale)
+    UI_APPEARANCE_SELECTOR;
+
+/** The text displayed in the text input. */
+@property(nonatomic, nullable, copy) NSString *text;
+
+/**
+ The presentation style of the text input.
+
+ Default is MDCTextInputPresentationStyleDefault.
+ */
+@property(nonatomic, assign)
+    MDCTextInputPresentationStyle presentationStyle NS_SWIFT_NAME(presentation);
+
+/**
+ The character limit for the text input. A label under the input counts characters entered and
+ presents the count / the limit.
+
+ Default is 0.
+ */
+@property(nonatomic, assign) IBInspectable NSUInteger characterLimit;
+
+/**
+ The color applied to the character count / character limit label.
+
+ Default is black with Material Design hint text opacity.
+ */
+@property(nonatomic, nullable, strong) UIColor *characterLimitColor NS_SWIFT_NAME(limitColor)
+    UI_APPEARANCE_SELECTOR;
+
+/**
+ The font applied to the character count / character limit label.
+
+ Default is system caption1.
+ */
+@property(nonatomic, nullable, strong) UIFont *characterLimitFont NS_SWIFT_NAME(limitFont)
+    UI_APPEARANCE_SELECTOR;
+
+/**
+ The color applied to the underline.
+
+ Default is black with Material Design hint text opacity.
+ */
+@property(nonatomic, nullable, strong) UIColor *underlineColor UI_APPEARANCE_SELECTOR;
+
+/** The thickness of the underline. */
+@property(nonatomic, assign) CGFloat underlineWidth UI_APPEARANCE_SELECTOR;
+
+/**
+ The text displayed under input rectangle. The Material Design guidelines call this 'Helper text.'
+
+ This is usually error text or further instructions.
```

Comment:
See MDCTextInputBehavior. That object has the previous simple API: setErrorText:errorAccessibilityValue